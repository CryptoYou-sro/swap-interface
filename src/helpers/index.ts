export * from './functions';
export * from './context';
export * from './constants';
export * from './auth';
export * from './graph';
export * from './storage';
export * from './types';
