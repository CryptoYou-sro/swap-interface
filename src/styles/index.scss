@import "./fonts/font-face.scss";
@import "./variables.scss";
