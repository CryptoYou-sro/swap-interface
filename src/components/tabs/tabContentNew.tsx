import { BLOCKS_AMOUNT, formatDate, makeId, routes, useStore } from '../../helpers';
import styled, { css } from 'styled-components';
import {
	DEFAULT_BORDER_RADIUS,
	DEFAULT_TRANSITION,
	fontSize,
	mediaQuery,
	pxToRem,
	spacing,
	Theme
} from '../../styles';
import { useBlockNumber } from '@usedapp/core';
import { useAxios } from '../../hooks';
import { useEffect, useState } from 'react';
import { Spinner } from '../spinner/spinner';

type Props = {
	data?: any;
	color?: string;
	toggleIndex?: number;
	type?: 'history' | 'swap';
};

type StyleProps = Props & { theme: Theme };

const Content = styled.div`
	color: ${(props: StyleProps) => props.theme.font.secondary};
	padding: ${spacing[20]};

	display: block;
	background: ${(props: StyleProps) => props.theme.background.secondary};
	border: 1px solid
		${(props: StyleProps) =>
			props.type === 'history' ? 'transparent' : props.theme.border.default};
	border-radius: ${DEFAULT_BORDER_RADIUS};
	border-top-left-radius: 0;
	margin-top: -1px;
	z-index: 10;
`;

export const ContentList = styled.ul`
	list-style: none;
	padding: 0;
`;

export const ContentItem = styled.li`
	list-style: none;
	padding: 0 0 ${spacing[26]} ${spacing[20]};
	border-left: 1px solid ${(props: StyleProps) => props.theme.font.default};
	position: relative;
	margin-left: ${spacing[10]};
	font-size: ${fontSize[16]};

	&:last-child {
		border: 0;
		padding-bottom: 0;
	}

	&:before {
		content: '';
		width: ${pxToRem(16)};
		height: ${pxToRem(16)};
		background: ${(props: StyleProps) => props.theme.font.default};
		border-radius: 50%;
		position: absolute;
		left: ${pxToRem(-8)};
		top: 0;

		${mediaQuery('s')} {
			width: ${pxToRem(14)};
			height: ${pxToRem(14)};
			left: ${pxToRem(-7)};
		}
	}

	&:last-child:before {
		background-color: ${(props: StyleProps) => props.color};
	}

	&:last-of-type > div {
		line-height: ${fontSize[16]};
	}
`;

export const ContentItemTitle = styled.div`
	line-height: ${fontSize[16]};
	margin-bottom: ${spacing[4]};
`;

export const ContentItemText = styled.div(() => {
	const {
		state: { theme }
	} = useStore();

	return css`
		color: ${(props: StyleProps) => (props.color ? props.color : theme.font.select)};
		line-height: ${fontSize[22]};
	`;
});

export const ContentItemLink = styled.div`
	color: ${(props: StyleProps) => props.theme.font.secondary};
	line-height: ${fontSize[16]};
	text-decoration: underline;
	cursor: pointer;
	transition: ${DEFAULT_TRANSITION};

	&:hover {
		color: ${(props: StyleProps) => props.theme.button.default};
	}
`;
const SpinnerWrapper = styled.div`
	display: flex;
	justify-content: center;

	${mediaQuery('s')} {
		margin-top: ${spacing[16]};
	}
`;

export const TabContentNew = ({ swap, type = 'swap' }: any) => {
	const [withdrawLink, setWithdrawLink] = useState<{
		amount: string;
		status: number;
		type: number;
		url: string | any;
		withdrawFee: string;
	} | null>(null);
	const currentBlockNumber = useBlockNumber();
	const {
		state: { theme }
	} = useStore();
	const orders = swap && swap.action[0];
	const withdrawal = swap && swap.withdraw[0];
	const api = useAxios();

	const getWithDrawLink = async () => {
		if (withdrawal) {
			try {
				await api
					.get(`${routes.transactionDetails}${withdrawal.id}`)
					.then((res) => setWithdrawLink(res.data));
			} catch (e) {
				console.log('e in getWithDrawLink function', { e });
			}
		}
	};
	useEffect(() => {
		void getWithDrawLink();
	}, [withdrawal]);

	return swap ? (
		// @ts-ignore
		<Content theme={theme} type={type}>
			<ContentList>
				{swap.costRequestCounter ? (
					<ContentItem theme={theme} key={makeId(32)}>
						<ContentItemTitle>
							Swap Request Validation ({swap.costRequestCounter}/2)
						</ContentItemTitle>
						<ContentItemText>
							{swap.costRequestCounter < 2
								? 'Your Swap request is under validation. Please wait until full confirmation.'
								: 'Your Swap request successfully validated.'}
						</ContentItemText>
					</ContentItem>
				) : null}
				{currentBlockNumber && swap.depositBlock ? (
					<ContentItem theme={theme} key={makeId(32)}>
						<ContentItemTitle>
							{!swap?.action.length
								? `Deposit confirmation (${
										currentBlockNumber - swap.depositBlock
								  }/${BLOCKS_AMOUNT})`
								: 'Deposit confirmed'}
						</ContentItemTitle>
						<ContentItemText>
							{currentBlockNumber - swap.depositBlock < BLOCKS_AMOUNT
								? 'Your deposit is waiting for the particular numbers of the blocks to pass. Please wait for 30 blocks to pass.'
								: currentBlockNumber - swap.depositBlock >= BLOCKS_AMOUNT && !swap.action.length
								? 'Your deposit is received and should be confirmed soon.'
								: null}
						</ContentItemText>
					</ContentItem>
				) : null}
				{orders ? (
					<ContentItem key={makeId(32)} theme={theme}>
						<ContentItemTitle>
							Conversion {swap.sourceToken} {orders.s.replace(swap.sourceToken, '')}
						</ContentItemTitle>
						<ContentItemText>Type: {orders.a === 0 ? 'SELL' : 'BUY'}</ContentItemText>
						<ContentItemText>Pair: {orders.s}</ContentItemText>
						<ContentItemText>Quantity: {orders.q}</ContentItemText>
						<ContentItemText>Price: {orders.p}</ContentItemText>
						<ContentItemText>Time: {formatDate(orders.ts)}</ContentItemText>
					</ContentItem>
				) : null}
				{withdrawal && !withdrawLink ? (
					<ContentItem key={makeId(32)} theme={theme}>
						<ContentItemLink theme={theme}>Withdrawal in process</ContentItemLink>
						<ContentItemText>
							Your funds is almost there, we are waiting for their landing into your wallet.
						</ContentItemText>
					</ContentItem>
				) : withdrawal?.t === 1 && withdrawLink ? (
					<ContentItem key={makeId(32)} theme={theme}>
						<ContentItemTitle>Withdrawal confirmed</ContentItemTitle>
					</ContentItem>
				) : withdrawal?.t === 0 && withdrawLink ? (
					<ContentItem key={makeId(32)} theme={theme}>
						<ContentItemLink theme={theme} onClick={() => window.open(withdrawLink.url)}>
							Withdrawal confirmed
						</ContentItemLink>
					</ContentItem>
				) : null}
				{swap.complete ? (
					<ContentItem theme={theme} color={theme.button.default}>
						<ContentItemText color={theme.button.default}>Successful swap!</ContentItemText>
					</ContentItem>
				) : !swap.complete && swap.complete !== null ? (
					<ContentItem theme={theme} color={theme.font.default}>
						<ContentItemText>No valid operations spotted!</ContentItemText>
					</ContentItem>
				) : null}
				{swap.complete === null ? (
					<SpinnerWrapper>
						<Spinner size="medium" color="default" />
					</SpinnerWrapper>
				) : null}
			</ContentList>
		</Content>
	) : null;
};
