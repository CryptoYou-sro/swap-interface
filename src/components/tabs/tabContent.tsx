import { BLOCKS_AMOUNT, useStore } from '../../helpers';
import { format } from 'date-fns';
import styled, { css } from 'styled-components';
import { pxToRem, spacing } from '../../styles';
import { useBlockNumber } from '@usedapp/core';

const Content = styled.div(() => {
	const {
		state: { theme }
	} = useStore();

	return css`
		color: ${theme.font.pure};
		padding: ${spacing[20]};
		display: block;
		background: ${theme.background.mobile};
		border: 1px solid ${theme.button.wallet};
	`;
});

const ContentList = styled.ul`
	list-style: none;
	padding: 0;
`;

const ContentItem = styled.li(({ color }: Props) => {
	const {
		state: { theme }
	} = useStore();

	return css`
		list-style: none;
		padding: 0 0 ${spacing[26]} ${spacing[20]};
		border-left: 1px solid ${theme.font.default};
		position: relative;
		margin-left: ${spacing[10]};

		&:last-child {
			border: 0;
			padding-bottom: 0;
			vertical-align: baseline;
		}

		&:last-child > p {
			margin: 0;
		}

		&:before {
			content: '';
			width: ${pxToRem(16)};
			height: ${pxToRem(16)};
			background: ${theme.font.default};
			border-radius: 50%;
			position: absolute;
			left: ${pxToRem(-8)};
			top: 0;
		}

		&:last-child:before {
			left: ${pxToRem(-8)};
			background-color: ${color};
		}
	`;
});

const ContentItemTitle = styled.h3`
	margin: 0;
`;

const ContentItemText = styled.p((props) => {
	return css`
		color: ${props.color};
	`;
});

const ContentItemLink = styled.a`
	text-decoration: underline;
	cursor: pointer;
`;
type Props = {
	data?: any;
	color?: string;
	toggle?: number;
};

export const TabContent = ({ data, toggle = 0 }: Props) => {
	const currentBlockNumber = useBlockNumber();
	const {
		state: { theme }
	} = useStore();
	const orders = data && data[toggle].action[0];
	const withdrawal = data && data[toggle].withdraw[0];
	const withdrawalLink: any = null;

	const headers = {
		Authorization: `Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiIweGZDZGIzM2JCNTlFZDhBMDFCOTBDYzEwOTgzMGFFZWVhY2Q1QWY3M2MiLCJ0eXBlIjoicmVmcmVzaCIsImV4cCI6MTY2NTgzNjM1NH0.3u4SubmpjncJ3kTEIE-QVDGSrlUlCPo55wTcnksqbT8
`,
		'Content-Type': 'application/json',
		'Access-Control-Allow-Origin': '*'
	};

	return (
		<Content>
			<ContentList>
				{data[toggle].costRequestCounter ? (
					<ContentItem key={Math.random()}>
						<ContentItemTitle>
							Swap Request Validation ({data[toggle].costRequestCounter}/2)
						</ContentItemTitle>
						<ContentItemText>
							{data[toggle].costRequestCounter < 2
								? 'Your Swap request is under validation. Please wait until full confirmation.'
								: 'Your Swap request successfully validated.'}
						</ContentItemText>
					</ContentItem>
				) : null}
				{currentBlockNumber && data[toggle].depositBlock ? (
					<ContentItem key={Math.random()}>
						<ContentItemTitle>
							{!data[toggle].action.length
								? `Deposit confirmation (${
										currentBlockNumber - data[toggle].depositBlock
								  }/${BLOCKS_AMOUNT})`
								: 'Deposit confirmed'}
						</ContentItemTitle>
						<ContentItemText>
							{currentBlockNumber - data[toggle].depositBlock < BLOCKS_AMOUNT
								? 'Your deposit is waiting for the particular numbers of the blocks to pass. Please wait for 30 blocks to pass.'
								: currentBlockNumber - data[toggle].depositBlock >= BLOCKS_AMOUNT &&
								  !data[toggle].action.length
								? 'Your deposit is received and should be confirmed soon.'
								: null}
						</ContentItemText>
					</ContentItem>
				) : null}
				{orders ? (
					<ContentItem key={Math.random()}>
						<ContentItemTitle>Conversion GLMR {orders.s.slice(4)}</ContentItemTitle>
						<ContentItemText>Type: {orders.a === 0 ? 'SELL' : 'BUY'}</ContentItemText>
						<ContentItemText>Pair: {orders.s}</ContentItemText>
						<ContentItemText>Quantity: {orders.q}</ContentItemText>
						<ContentItemText>Price: {orders.p}</ContentItemText>
						<ContentItemText>
							Time: {format(new Date(orders.ts * 1000), 'dd/MM/yyyy kk:mm:ss')}
						</ContentItemText>
					</ContentItem>
				) : null}
				{withdrawal && !withdrawalLink ? (
					<ContentItem key={Math.random()}>
						<ContentItemLink>Withdrawal in progress</ContentItemLink>
						<ContentItemText>
							Your funds is almost there, we are waiting for their landing into your wallet.
						</ContentItemText>
					</ContentItem>
				) : withdrawalLink ? (
					<ContentItem key={Math.random()}>
						<ContentItemLink href={withdrawalLink.url}>Withdrawal confirmed</ContentItemLink>
					</ContentItem>
				) : null}
				{data[toggle].complete === true ? (
					<ContentItem key={Math.random()} color={theme.button.default}>
						<ContentItemText color={theme.button.default}>Successful swap!</ContentItemText>
					</ContentItem>
				) : null}
				{data[toggle].complete === null ? (
					<ContentItem key={Math.random()} color={theme.font.pure}>
						<ContentItemText color={theme.font.pure}>Not valid operations spotted!</ContentItemText>
					</ContentItem>
				) : null}
			</ContentList>
		</Content>
	);
};
