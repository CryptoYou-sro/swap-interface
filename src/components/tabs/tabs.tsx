import { format } from 'date-fns';
import styled, { css } from 'styled-components';
import { BLOCKS_AMOUNT, useStore } from '../../helpers';
import { useState } from 'react';
import { DEFAULT_BORDER_RADIUS, pxToRem, spacing } from '../../styles';
import { useBlockNumber } from '@usedapp/core';

const Wrapper = styled.div`
	display: flex;
	flex-direction: column;
	max-width: 100%;
`;

const TabsContainer = styled.div`
	display: flex;
	overflow-x: auto;
`;

const Tab = styled.div(({ active }: Props) => {
	const {
		state: { theme }
	} = useStore();

	return css`
		z-index: 10;
		cursor: pointer;
		color: ${theme.font.pure};
		padding: ${spacing[6]} ${spacing[6]};
		text-align: center;
		margin-right: ${spacing[4]};
		background: ${theme.background.mobile};
		border-radius: ${DEFAULT_BORDER_RADIUS} ${DEFAULT_BORDER_RADIUS} 0 0;
		border: 1px solid ${theme.button.wallet};
		border-bottom: none;

		&:nth-child(${++active}) {
			border-bottom: 1px solid ${theme.button.default};
		}

		&:last-child {
			margin-right: 0;
		}
	`;
});

const Content = styled.div(() => {
	const {
		state: { theme }
	} = useStore();

	return css`
		color: ${theme.font.pure};
		padding: ${spacing[20]};
		display: block;
		background: ${theme.background.mobile};
		border: 1px solid ${theme.button.wallet};
	`;
});

const ContentList = styled.ul`
	list-style: none;
	padding: 0;
`;

const ContentItem = styled.li(({ color }: Props) => {
	const {
		state: { theme }
	} = useStore();

	return css`
		list-style: none;
		padding: 0 0 ${spacing[26]} ${spacing[20]};
		border-left: 1px solid ${theme.font.default};
		position: relative;
		margin-left: ${spacing[10]};

		&:last-child {
			border: 0;
			padding-bottom: 0;
			vertical-align: baseline;
		}

		&:last-child > p {
			margin: 0;
		}

		&:before {
			content: '';
			width: ${pxToRem(13)};
			height: ${pxToRem(13)};
			background: ${theme.font.default};
			border-radius: 50%;
			position: absolute;
			left: ${pxToRem(-7)};
			top: 0;
		}

		&:last-child:before {
			left: ${pxToRem(-6)};
			background-color: ${color ? theme.button.default : theme.font.default};
		}
	`;
});

const ContentItemTitle = styled.h3`
	margin: 0;
`;

const ContentItemText = styled.p((props) => {
	return css`
		color: ${props.color};
	`;
});

const ContentItemLink = styled.a`
	text-decoration: underline;
`;

type Props = {
	data?: any;
	color?: boolean;
	active?: any;
};

export const Tabs = ({ data }: Props) => {
	const currentBlockNumber = useBlockNumber();
	const {
		state: { destinationToken, theme }
	} = useStore();
	const [toggle, setToggle] = useState(0);
	const handleToggle = (index: number) => setToggle(index);
	const orders = data && data[toggle].action;

	return (
		<Wrapper>
			{data?.length > 0 && (
				<>
					<TabsContainer>
						{data?.length &&
							data.map((item: any) => {
								const index = data.indexOf(item);

								return (
									<Tab key={Math.random()} onClick={() => handleToggle(index)} active={toggle}>
										GLMR {destinationToken}
									</Tab>
								);
							})}
					</TabsContainer>
					<Content>
						<ContentList>
							{data[toggle].costRequestCounter ? (
								<ContentItem key={Math.random()}>
									<ContentItemTitle>
										Swap Request Validation ({data[toggle].costRequestCounter}/2)
									</ContentItemTitle>
									<ContentItemText>
										{data[toggle].costRequestCounter === 1
											? 'Your Swap request is under validation. Please wait until full confirmation.'
											: 'Your Swap request successfully validated.'}
									</ContentItemText>
								</ContentItem>
							) : null}
							{currentBlockNumber && data[toggle].depositBlock ? (
								<ContentItem key={Math.random()}>
									<ContentItemTitle>
										{!data[toggle].action.length
											? `Deposit confirmation (${
													currentBlockNumber - data[toggle].depositBlock
											  }/${BLOCKS_AMOUNT})`
											: 'Deposit confirmed'}
									</ContentItemTitle>
									<ContentItemText>
										{data[toggle].depositBlock < BLOCKS_AMOUNT
											? 'Your deposit is waiting for the particular numbers of the blocks to pass. Please wait for 30 blocks to pass.'
											: !data[toggle].action
											? 'Your deposit is received and should be confirmed soon.'
											: null}
									</ContentItemText>
								</ContentItem>
							) : null}
							{orders?.length
								? orders.map((item: any) => {
										return (
											<ContentItem key={Math.random()}>
												<ContentItemTitle>Conversion GLMR {item.s.slice(4)}</ContentItemTitle>
												<ContentItemText>Type: {item.t === 1 ? 'SELL' : 'BUY'}</ContentItemText>
												<ContentItemText>Pair: {item.s}</ContentItemText>
												<ContentItemText>Quantity: {item.q.replace(/0*$/, '')}</ContentItemText>
												<ContentItemText>Price: {item.p}</ContentItemText>
												<ContentItemText>
													Time: {format(new Date(item.ts * 1000), 'dd/MM/yyyy kk:mm:ss')}
												</ContentItemText>
											</ContentItem>
										);
								  })
								: null}
							{data[toggle].withdraw && (
								<ContentItem key={Math.random()}>
									<ContentItemLink>Withdrawal confirmed!</ContentItemLink>
									<ContentItemText>You can check transaction by link</ContentItemText>
								</ContentItem>
							)}
							{data[toggle].complete === true ? (
								<ContentItem key={Math.random()} color={data[toggle].complete}>
									<ContentItemText
										color={data[toggle].complete ? theme.button.default : theme.font.pure}>
										Successful swap!
									</ContentItemText>
								</ContentItem>
							) : null}
							{data[toggle].complete === null ? (
								<ContentItem key={Math.random()} color={data[toggle].complete}>
									<ContentItemText
										color={data[toggle].complete ? theme.button.default : theme.font.pure}>
										Not valid operations spotted!
									</ContentItemText>
								</ContentItem>
							) : null}
						</ContentList>
					</Content>
				</>
			)}
		</Wrapper>
	);
};
