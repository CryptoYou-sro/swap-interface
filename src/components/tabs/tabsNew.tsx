import styled, { css } from 'styled-components';
import { useStore } from '../../helpers';
import { DEFAULT_BORDER_RADIUS, spacing } from '../../styles';
import { useState } from 'react';
import { TabContentNew } from './tabContentNew';

const Wrapper = styled.div`
	display: flex;
	flex-direction: column;
	max-width: 100%;
`;

const TabsContainer = styled.div`
	display: flex;
	overflow-x: auto;
`;

const Tab = styled.div(({ active }: Props) => {
	const {
		state: { theme }
	} = useStore();

	return css`
		cursor: pointer;
		color: ${theme.font.secondary};
		padding: ${spacing[6]} ${spacing[6]};
		text-align: center;
		margin-right: ${spacing[4]};
		background: ${theme.background.secondary};
		border-radius: ${DEFAULT_BORDER_RADIUS} ${DEFAULT_BORDER_RADIUS} 0 0;
		border: 1px solid ${theme.border.default};
		border-bottom: ${active ? `1px solid ${theme.button.default}` : 'none'};

		&:last-child {
			margin-right: 0;
		}
	`;
});

type Props = {
	active: boolean;
};

export const TabsNew = ({ swap }: any) => {
	const [active, setActive] = useState(false);
	console.log(swap);

	return (
		<Wrapper>
			<TabsContainer>
				<Tab active={active} onClick={() => setActive(true)}>
					{swap.pair}
				</Tab>
			</TabsContainer>
			<TabContentNew swap={swap} selectItem={active} />
		</Wrapper>
	);
};
