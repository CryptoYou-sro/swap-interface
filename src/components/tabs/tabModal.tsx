import { Tabs } from './tabs';
import styled from 'styled-components';
import { pxToRem } from '../../styles';

const Wrapper = styled.div`
	max-width: 100%;
	margin: ${pxToRem(76)} auto;
`;

const Paragraph = styled.p`
	color: #b4b4b4;
`;

export const TabModal = () => {
	return (
		<Wrapper>
			<Paragraph>Pending Swaps ()</Paragraph>
			<Tabs
				data={[
					{
						id: 0,
						costRequestCounter: 1,
						depositBlock: 10,
						action: [
							{
								t: 0,
								id: '39ff037641ab4658b4f1d2831e9132ce'
							},
							{
								t: 0,
								a: 1,
								s: 'GLMRBUSD',
								q: 24.0,
								p: 0.8503,
								ts: 1654846854
							}
						],
						withdraw: true,
						complete: true
					},
					{
						id: 0,
						costRequestCounter: 1,
						depositBlock: 10,
						action: [
							{
								t: 0,
								id: '39ff037641ab4658b4f1d2831e9132ce'
							},
							{
								t: 0,
								a: 0,
								s: 'GLMRBTC',
								q: 95.0,
								p: 0.7907,
								ts: 1654006400
							}
						],
						withdraw: true,
						complete: null
					}
				]}
			/>
		</Wrapper>
	);
};
