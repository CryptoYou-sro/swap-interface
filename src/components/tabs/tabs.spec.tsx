import 'jest-styled-components';
import { render } from '@testing-library/react';
import { AuthProvider } from '../../helpers';
import { Tabs } from './tabs';
import { darkTheme, DEFAULT_BORDER_RADIUS, spacing } from '../../styles';
import { format } from 'date-fns';

describe('SelectList', () => {
	it('Render tab component with one tab', function () {
		const eventsData = [
			{
				id: 0,
				costRequestCounter: 1,
				depositBlock: 10,
				action: [
					{
						t: 0,
						a: 1,
						s: 'GLMRBUSD',
						q: 24.0,
						p: 0.8503,
						ts: 1654846854
					}
				],
				withdraw: true,
				complete: true
			}
		];

		const { getByText, getByRole, getByTestId, getAllByRole } = render(
			<AuthProvider>
				<Tabs data={eventsData} />
			</AuthProvider>
		);

		const wrapper = getByTestId('tabs-container');
		expect(wrapper).toBeInTheDocument();
		expect(wrapper).toMatchSnapshot();
		expect(wrapper).toHaveStyle(`
			display: flex;
			flex-direction: column;
			max-width: 100%;
			`);

		const Tab = getByText('GLMR Select Token');
		expect(Tab).toBeInTheDocument();
		expect(Tab).toHaveStyle(
			`
		z-index: 10;
		cursor: pointer;
		color: ${darkTheme.font.secondary};
		padding: ${spacing[6]} ${spacing[6]};
		text-align: center;
		margin-right: 0;
		background: ${darkTheme.background.secondary};
		border-radius: ${DEFAULT_BORDER_RADIUS} ${DEFAULT_BORDER_RADIUS} 0 0;
		border: 1px solid ${darkTheme.button.wallet};
		border-bottom: 1px solid ${darkTheme.button.default};
			`
		);

		const contentList = getByRole('list');
		expect(contentList).toBeInTheDocument();
		expect(contentList).toHaveStyle(`
		list-style: none;
		padding: 0;
		`);

		const contentListItems = getAllByRole('listitem');
		expect(contentListItems.length).toBe(4);

		const contentItemTitleCostRequest = getByText(
			`Swap Request Validation (${eventsData[0].costRequestCounter}/2)`
		);
		expect(contentItemTitleCostRequest).toBeInTheDocument();
		expect(contentItemTitleCostRequest).toHaveStyle('margin: 0');

		const contentItemTitleDeposit = getByText(
			`Conversion GLMR ${eventsData[0].action[0].s.slice(4)}`
		);
		expect(contentItemTitleDeposit).toBeInTheDocument();

		const itemTextCostRequest = getByText(
			'Your Swap request is under validation. Please wait until full confirmation.'
		);
		expect(itemTextCostRequest).toBeInTheDocument();

		const itemTextType = getByText('Type: BUY');
		expect(itemTextType).toBeInTheDocument();

		const itemTextPair = getByText(`Pair: ${eventsData[0].action[0].s}`);
		expect(itemTextPair).toBeInTheDocument();

		const itemTextQuantity = getByText(`Quantity: ${eventsData[0].action[0].q}`);
		expect(itemTextQuantity).toBeInTheDocument();

		const itemTextPrice = getByText(`Price: ${eventsData[0].action[0].p}`);
		expect(itemTextPrice).toBeInTheDocument();

		const itemTextTime = getByText(
			`Time: ${format(new Date(eventsData[0].action[0].ts * 1000), 'dd/MM/yyyy kk:mm:ss')}`
		);
		expect(itemTextTime).toBeInTheDocument();

		const itemTextStatus = getByText('Successful swap!');
		expect(itemTextStatus).toBeInTheDocument();
		expect(itemTextStatus).toHaveStyle(`color: ${darkTheme.button.default}`);
	});

	it.only('Render tab component with correct length, data and styles', function () {
		const eventsData = [
			{
				id: 0,
				costRequestCounter: 2,
				depositBlock: 10,
				action: [
					{
						t: 0,
						a: 1,
						s: 'GLMRBUSD',
						q: 24.0,
						p: 0.8503,
						ts: 1654846854
					}
				],
				withdraw: true,
				complete: false
			},
			{
				id: 0,
				costRequestCounter: 1,
				depositBlock: 30,
				action: [
					{
						t: 0,
						a: 1,
						s: 'GLMRBTC',
						q: 24.0,
						p: 0.8503,
						ts: 1654846854
					}
				],
				withdraw: true,
				complete: false
			}
		];

		const { getByTestId } = render(
			<AuthProvider>
				<Tabs data={eventsData} />
			</AuthProvider>
		);

		const wrapper = getByTestId('tabs-container');
		expect(wrapper).toBeInTheDocument();
		expect(wrapper).toMatchSnapshot();
		expect(wrapper).toHaveStyle(`
			display: flex;
			flex-direction: column;
			max-width: 100%;
			`);

		const TabsContainer = getByTestId('tabs');
		expect(TabsContainer).toBeInTheDocument();
	});
});
