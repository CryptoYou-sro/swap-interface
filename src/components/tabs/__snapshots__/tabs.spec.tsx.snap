// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectList Render tab component with one tab, content with correct data and styles 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow-x: auto;
}

.c2 {
  z-index: 10;
  cursor: pointer;
  color: #FFF;
  padding: 0.375rem 0.375rem;
  text-align: center;
  margin-right: 0.25rem;
  background: #1C2125;
  border-radius: 0.375rem 0.375rem 0 0;
  border: 1px solid #505050;
  border-bottom: none;
}

.c2:nth-child(1) {
  border-bottom: 1px solid #00A8E8;
}

.c2:last-child {
  margin-right: 0;
}

.c3 {
  color: #FFF;
  padding: 1.25rem;
  display: block;
  background: #1C2125;
  border: 1px solid #505050;
}

.c4 {
  list-style: none;
  padding: 0;
}

.c5 {
  list-style: none;
  padding: 0 0 1.625rem 1.25rem;
  border-left: 1px solid #8C8D8F;
  position: relative;
  margin-left: 0.625rem;
}

.c5:last-child {
  border: 0;
  padding-bottom: 0;
  vertical-align: baseline;
}

.c5:last-child > p {
  margin: 0;
}

.c5:before {
  content: '';
  width: 0.8125rem;
  height: 0.8125rem;
  background: #8C8D8F;
  border-radius: 50%;
  position: absolute;
  left: -0.4375rem;
  top: 0;
}

.c5:last-child:before {
  left: -0.375rem;
}

.c8 {
  list-style: none;
  padding: 0 0 1.625rem 1.25rem;
  border-left: 1px solid #8C8D8F;
  position: relative;
  margin-left: 0.625rem;
}

.c8:last-child {
  border: 0;
  padding-bottom: 0;
  vertical-align: baseline;
}

.c8:last-child > p {
  margin: 0;
}

.c8:before {
  content: '';
  width: 0.8125rem;
  height: 0.8125rem;
  background: #8C8D8F;
  border-radius: 50%;
  position: absolute;
  left: -0.4375rem;
  top: 0;
}

.c8:last-child:before {
  left: -0.375rem;
  background-color: #00A8E8;
}

.c6 {
  margin: 0;
}

.c9 {
  color: #00A8E8;
}

.c7 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div
  class="c0"
  data-testid="tabs-container"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      GLMR 
      Select Token
    </div>
  </div>
  <div
    class="c3"
  >
    <ul
      class="c4"
    >
      <li
        class="c5"
      >
        <h3
          class="c6"
        >
          Swap Request Validation (
          1
          /2)
        </h3>
        <p
          class=""
        >
          Your Swap request is under validation. Please wait until full confirmation.
        </p>
      </li>
      <li
        class="c5"
      >
        <h3
          class="c6"
        >
          Conversion GLMR 
          BUSD
        </h3>
        <p
          class=""
        >
          Type: 
          BUY
        </p>
        <p
          class=""
        >
          Pair: 
          GLMRBUSD
        </p>
        <p
          class=""
        >
          Quantity: 
          24
        </p>
        <p
          class=""
        >
          Price: 
          0.8503
        </p>
        <p
          class=""
        >
          Time: 
          10/06/2022 10:40:54
        </p>
      </li>
      <li
        class="c5"
      >
        <a
          class="c7"
        >
          Withdrawal confirmed!
        </a>
        <p
          class=""
        >
          You can check transaction by link
        </p>
      </li>
      <li
        class="c8"
        color="#00A8E8"
      >
        <p
          class="c9"
          color="#00A8E8"
        >
          Successful swap!
        </p>
      </li>
    </ul>
  </div>
</div>
`;
